src/plugin.cpp

```diff
@@ -1,22 +1,58 @@
 #define IMPLEMENTS_GLOBALS
 #include <configuration.h>
 #include <core.h>
-#include <dicom-file.h>
+#include <dicom-tag.h>
 
 #include <nlohmann/json.hpp>
 
 namespace nlm = nlohmann;
+namespace fs = std::filesystem;
 namespace globals {
     OrthancPluginContext *context = nullptr;
     TagFilter filter_list;
+    std::string storage_location;
+    fs::perms dir_permissions = fs::perms::owner_all | fs::perms::group_all | fs::perms::others_read | fs::perms::sticky_bit;
+    fs::perms file_permissions = fs::perms::owner_all | fs::perms::group_all | fs::perms::others_read;
 }
 
-// prototypes
-int32_t FilterCallback(const OrthancPluginDicomInstance* instance);
-void PopulateFilterList();
+void PopulateFilterList(const nlm::json &config){
+    // iterate the Dicom-Filter configuration tags array
+    for (const auto &iter: config["Dicom-Filter"]["tags"]) {
+        // todo: check that the string has 9 characters; true: do below, false: implement full group filters (eg. "0002,*", "0002")
+        // get tag string, convert to decimal
+        auto tag = iter.get<std::string>();
+        tag.append(tag.substr(0, 4));
+        tag.erase(0, 5);
+        uint32_t tag_code = HexToDec(tag);
+        // register tag
+        globals::filter_list.emplace(tag_code);
```

Comment:
If we create a class for TagFilters, then we can [more] easily add different forms of filters like full groups.